dot/parachain/availability-store/availability_store_test.go

```diff
@@ -1752,3 +1795,144 @@ func TestStoredDataKeptUntilFinalized(t *testing.T) {
+	require.NoError(harness.t, err)
+	candidate1Events.Add(parachaintypes.CandidateIncluded{CandidateReceipt: candidate1})
+
+	blockNumber2 := uint32(5)
+	parent2 := common.Hash{0x04, 0x04, 0x04, 0x04}
+	pov2 := parachaintypes.PoV{BlockData: parachaintypes.BlockData{4, 5, 6}}
+	pov2Hash := common.MustBlake2bHash(scale.MustMarshal(pov2))
+	candidate2 := buildCandidateReceipt(pov2Hash)
+	candidate2Hash, err := candidate2.Hash()
+	require.NoError(t, err)
+	availableData2 := AvailableData{
+		PoV: pov2,
+		ValidationData: parachaintypes.PersistedValidationData{
+			ParentHead: parachaintypes.HeadData{Data: []byte{}},
+		},
+	}
+	availableData2ErasureRoot := buildAvailableDataBranchesRoot(t, nValidators, availableData2)
+	msg2StoreAvailabeData := StoreAvailableData{
+		CandidateHash:       parachaintypes.CandidateHash{Value: candidate2Hash},
+		NumValidators:       nValidators,
+		AvailableData:       availableData2,
+		ExpectedErasureRoot: availableData2ErasureRoot,
+		Sender:              msgSenderChan,
+	}
+	candidate2Events, err := parachaintypes.NewCandidateEvents()
+	require.NoError(harness.t, err)
+	candidate2Events.Add(parachaintypes.CandidateIncluded{CandidateReceipt: candidate2})
+
+	err = harness.overseer.Start()
+	require.NoError(t, err)
+
+	go harness.processMessages()
+
+	harness.broadcastMessages = append(harness.broadcastMessages, msg1StoreAvailableData)
+	harness.triggerBroadcast()
+
+	// result from seeding data
+	msg1SenderChanResult := <-msg1StoreAvailableData.Sender
+	require.Equal(t, nil, msg1SenderChanResult)
+
+	harness.broadcastMessages = append(harness.broadcastMessages, msg2StoreAvailabeData)
+	harness.triggerBroadcast()
+
+	// result from seeding data 2
+	msgSender2ChanResult := <-msg2StoreAvailabeData.Sender
+	require.Equal(t, nil, msgSender2ChanResult)
+
+	// confirm available data 1 and 2, and has all chunks 1 and 2
+	availableDataResult := harness.queryAvailableData(parachaintypes.CandidateHash{Value: candidate1Hash})
+	require.Equal(t, availableData1, availableDataResult)
+	availableDataResult = harness.queryAvailableData(parachaintypes.CandidateHash{Value: candidate2Hash})
+	require.Equal(t, availableData2, availableDataResult)
+	hasChunks := harness.hasAllChunks(parachaintypes.CandidateHash{Value: candidate1Hash}, nValidators, true)
+	require.True(t, hasChunks)
+	hasChunks = harness.hasAllChunks(parachaintypes.CandidateHash{Value: candidate2Hash}, nValidators, true)
+	require.True(t, hasChunks)
+	harness.printDB("before import leaf")
+
+	// import leaf for candidate 1
+	aLeaf := harness.importLeaf(t, parent1, blockNumber1, candidate1Events)
+	time.Sleep(50 * time.Millisecond)
+
+	// import leaf for candidate 2
+	harness.importLeaf(t, parent2, blockNumber2, candidate2Events)
+	time.Sleep(1500 * time.Millisecond)
+	harness.printDB("after import leaf")
+
+	// signal block 1 finalized for candidate 1
+	blockFinalizedSignal := parachaintypes.BlockFinalizedSignal{
+		Hash:        aLeaf,
```

Comment:
lets use activatedLeaf instead of aLeaf?

Reply:
Updated.